### PR TITLE
A/B test: FacebookLikePrompt

### PR DESCRIPTION
**Pull request by OliverJAsh**
# #ab-FacebookLikePrompt=progress-bar--simple

Mobile
image

Desktop
image
# #ab-FacebookLikePrompt=progress-bar--high-target

Mobile
image

Desktop
image
# #ab-FacebookLikePrompt=progress-bar--low-target

Mobile
image

Desktop
image
## 

Although I have only provided 6 screenshots, there are actually 27 different combinations: `breakpoints (~3) * test variants (3) * Facebook pages (1 per edition = 3)`. Getting the numbers to fit inside of the progress bar is quite fiddly. I attempted a simple solution but in the end it turned out pretty complicated because there are so many combinations.

So, forgive me if I missed an edge case here, but I couldn't stomach waiting for the CSS to compile any more. :unamused: 

I have tested my CSS changes against the international edition site message.

TODO:
- [ ] Look into Facebook Graph API rate limiting (we are making a request for each user)

/cc @crifmulholland

### CODE REVIEW

File: static/src/javascripts/projects/common/modules/experiments/tests/facebook-like-prompt.js [hunk 1]
```diff
@@ -0,0 +1,176 @@
...
+                .then(function (response) {
+                    return response.likes;
+                });
+        };
+
+        var loadFacebookSdk = function () {
+            // Render like button
+            // SDK must be loaded/initialised after the like button is added
+
+            // TODO: Why do we have to initialise twice? Test on mobile + desktop!!
```
Comment (OliverJAsh) [inline static/src/javascripts/projects/common/modules/experiments/tests/facebook-like-prompt.js]: The FB SDK might already be loaded and initialised by `FacebookAuthorizer`. I had trouble detecting this, so in the end I ended up initialising the SDK again. FB SDK gives us a warning about double initialising.

image

File: static/src/javascripts/projects/common/modules/experiments/tests/facebook-like-prompt.js [hunk 2]
```diff
@@ -0,0 +1,176 @@
...
+                    'US': { target: 500000 }
+                }, { showIncompleteLabelInsideProgressBar: true })
+            },
+            {
+                id: 'progress-bar--low-target',
+                test: progressBarTest({
+                    'UK': { target: 5000000, textTarget: '5 million' },
+                    'AU': { target: 400000 },
+                    'US': { target: 300000 }
+                })
```
Comment (OliverJAsh) [inline static/src/javascripts/projects/common/modules/experiments/tests/facebook-like-prompt.js]: @crifmulholland What do you think to these targets for the low and high variants?

Comment (crifmulholland) [inline static/src/javascripts/projects/common/modules/experiments/tests/facebook-like-prompt.js]: Lets do this:

US is a bit more extreme, but there's certainly more growth potential there.

{
+                id: 'progress-bar--high-target',
+                test: progressBarTest({
+                    'UK': { target: 10000000, textTarget: '10 million' },
+                    'AU': { target: 750000 },
+                    'US': { target: 1000000 }
+                }, { showIncompleteLabelInsideProgressBar: true })
+            },
+            {
+                id: 'progress-bar--low-target',
+                test: progressBarTest({
+                    'UK': { target: 5000000, textTarget: '5 million' },
+                    'AU': { target: 300000 },
+                    'US': { target: 250000 }
+                })

Comment (OliverJAsh) [inline static/src/javascripts/projects/common/modules/experiments/tests/facebook-like-prompt.js]: Is it a fair test if the US target is higher than AU for the high target
variant, but vice versa for the low target variant?

On Wed, 22 Apr 2015 at 10:54 Chris Mulholland <notifications@github.com>
wrote:

> In
> static/src/javascripts/projects/common/modules/experiments/tests/facebook-like-prompt.js
> <https://github.com/guardian/frontend/pull/8958#discussion_r28858402>:
>
> > +            },
> > +            {
> > +                id: 'progress-bar--high-target',
> > +                test: progressBarTest({
> > +                    'UK': { target: 10000000, textTarget: '10 million' },
> > +                    'AU': { target: 500000 },
> > +                    'US': { target: 500000 }
> > +                }, { showIncompleteLabelInsideProgressBar: true })
> > +            },
> > +            {
> > +                id: 'progress-bar--low-target',
> > +                test: progressBarTest({
> > +                    'UK': { target: 5000000, textTarget: '5 million' },
> > +                    'AU': { target: 400000 },
> > +                    'US': { target: 300000 }
> > +                })
>
> Lets do this:
>
> US is a bit more extreme, but there's certainly more growth potential
> there.
>
> {
>
>    - id: 'progress-bar--high-target',
>    - test: progressBarTest({
>    - 'UK': { target: 10000000, textTarget: '10 million' },
>    - 'AU': { target: 750000 },
>    - 'US': { target: 1000000 }
>    - }, { showIncompleteLabelInsideProgressBar: true })
>    - },
>    - {
>    - id: 'progress-bar--low-target',
>    - test: progressBarTest({
>    - 'UK': { target: 5000000, textTarget: '5 million' },
>    - 'AU': { target: 300000 },
>    - 'US': { target: 250000 }
>    - })
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/guardian/frontend/pull/8958/files#r28858402>.
>

-- 

------------------------------
Visit theguardian.com. On your mobile and tablet, download the Guardian 
iPhone and Android apps theguardian.com/guardianapp and our tablet editions 
theguardian.com/editions.  Save up to 57% by subscribing to the Guardian 
and Observer - choose the papers you want and get full digital access. 
 Visit subscribe.theguardian.com

This e-mail and all attachments are confidential and may also be 
privileged. If you are not the named recipient, please notify the sender 
and delete the e-mail and all attachments immediately. Do not disclose the 
contents to another person. You may not use the information for any 
purpose, or store, or copy, it in any way.  Guardian News & Media Limited 
is not liable for any computer viruses or other material transmitted with 
or as part of this e-mail. You should employ virus checking software.
 
Guardian News & Media Limited is a member of Guardian Media Group plc. Registered 
Office: PO Box 68164, Kings Place, 90 York Way, London, N1P 2AP.  Registered 
in England Number 908396

**Comment by crifmulholland** [comment]
I'm still worried about tracking around this. It looks like facebook offers a callback when the button is clicked. 

https://developers.facebook.com/docs/reference/javascript/FB.Event.subscribe/v2.3 

May be a different pull request, but can we have a look please?

**Comment by OliverJAsh** [comment]
Nice find. I should be able to add this fairly easily. My guess is we only
care for tracking likes, not unlikes?

On Wed, 22 Apr 2015 at 12:00 Chris Mulholland <notifications@github.com>
wrote:

> I'm still worried about tracking around this. It looks like facebook
> offers a callback when the button is clicked.
>
>
> https://developers.facebook.com/docs/reference/javascript/FB.Event.subscribe/v2.3
>
> May be a different pull request, but can we have a look please?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/guardian/frontend/pull/8958#issuecomment-95137885>.
>

-- 

------------------------------
Visit theguardian.com. On your mobile and tablet, download the Guardian 
iPhone and Android apps theguardian.com/guardianapp and our tablet editions 
theguardian.com/editions.  Save up to 57% by subscribing to the Guardian 
and Observer - choose the papers you want and get full digital access. 
 Visit subscribe.theguardian.com

This e-mail and all attachments are confidential and may also be 
privileged. If you are not the named recipient, please notify the sender 
and delete the e-mail and all attachments immediately. Do not disclose the 
contents to another person. You may not use the information for any 
purpose, or store, or copy, it in any way.  Guardian News & Media Limited 
is not liable for any computer viruses or other material transmitted with 
or as part of this e-mail. You should employ virus checking software.
 
Guardian News & Media Limited is a member of Guardian Media Group plc. Registered 
Office: PO Box 68164, Kings Place, 90 York Way, London, N1P 2AP.  Registered 
in England Number 908396

File: static/src/javascripts/projects/common/modules/experiments/tests/facebook-like-prompt.js [hunk 3]
```diff
@@ -0,0 +1,179 @@
...
+    detect,
+    template,
+    Message,
+    config,
+    ajax
+) {
+
+    return function () {
+        this.id = 'FacebookLikePrompt';
+        this.start = '2015-04-17';
```
Comment (rich-nguyen) [inline static/src/javascripts/projects/common/modules/experiments/tests/facebook-like-prompt.js]: update

**Comment by rich-nguyen** [comment]
:+1:

**Pull request merged**